shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/impl/OracleDMLStatementSQLVisitor.java

```diff
@@ -487,6 +492,15 @@ public ASTNode visitMerge(final MergeContext ctx) {
         result.setTarget((SimpleTableSegment) visit(ctx.intoClause()));
         result.setSource((TableSegment) visit(ctx.usingClause()));
         result.setExpr((ExpressionSegment) (visit(ctx.usingClause().expr())));
+        if (null != ctx.mergeUpdateClause()) {
+            result.getUpdate().setSetAssignment((SetAssignmentSegment) visit(ctx.mergeUpdateClause().mergeSetAssignmentsClause()));
+            if (null != ctx.mergeUpdateClause().whereClause()) {
+                result.getUpdate().setWhere((WhereSegment) visit(ctx.mergeUpdateClause().whereClause()));
+            }
+            if (null != ctx.mergeUpdateClause().deleteWhereClause()) {
+                result.getUpdate().setWhere((WhereSegment) visit(ctx.mergeUpdateClause().deleteWhereClause()));
```

Comment:
@tristaZero @ThanoshanMV  Delete Where is not suitable in the update statement. Maybe should put in a delete statement.

Reply:
Okay @wgy8283335, I'll put `DeleteStatement` as a member in the `MergeStatement`. So that we can visit `deleteWhereClause` and put it inside `DeleteStatement`'s `WhereSegment`.

Reply:
@wgy8283335 since we get `WhereSegement` actual values from two statements: `UpdateStatement` and `DeleteStatement`, to expect and assert those `WhereSegement` values, we should have two different `ExpectedWhereClause` right? 

Can we have something like below in the `MergeStatementTestCase`? :

![MergeStatementTestCase](https://user-images.githubusercontent.com/48581379/112149565-247f9c00-8c05-11eb-87cc-8669016a0e5a.png)

Reply:
Yes, I think so.

Reply:
@wgy8283335 I added `DeleteStatement` as a member in the `MergeStatement` and modified the `MergeStatementTestCase` like above. 

I asserted both `WhereSegements` inside `assertWhereClause` method in `MergeStatementAssert`.